solidity/test/BancorNetworkPathFinder.js

```diff
@@ -4,18 +4,81 @@
 const BancorConverter = artifacts.require('BancorConverter');
+const BancorConverterFactory = artifacts.require('BancorConverterFactory');
 const BancorConverterRegistry = artifacts.require('BancorConverterRegistry');
 const BancorConverterRegistryData = artifacts.require('BancorConverterRegistryData');
 const BancorNetworkPathFinder = artifacts.require('BancorNetworkPathFinder');
 
-async function print(sourceToken, targetToken, path) {
-    const sourceSymbol = await SmartToken.at(sourceToken).symbol();
-    const targetSymbol = await SmartToken.at(targetToken).symbol();
-    const symbols = await Promise.all(path.map(token => SmartToken.at(token).symbol()));
+const ETH_RESERVE_ADDRESS = '0x'.padEnd(42, 'e');
+
+const ANCHOR_TOKEN_SYMBOL = 'ETH';
+
+const layout = {
+    'reserves': [
+        {'symbol': 'AAA', 'decimals': 18, 'supply': '1e24'},
+        {'symbol': 'BBB', 'decimals': 18, 'supply': '1e24'},
+        {'symbol': 'CCC', 'decimals': 18, 'supply': '1e24'},
+        {'symbol': 'DDD', 'decimals': 18, 'supply': '1e24'},
+    ],
+    'converters': [
+        {
+            'symbol': 'BNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'ETH', 'weight': 500000, 'balance': '1e24'},
+            ]
+        },
+        {
+            'symbol': 'AAABNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'AAA', 'weight': 500000, 'balance': '1e21'},
+                {'symbol': 'BNT', 'weight': 500000, 'balance': '1e21'},
+            ]
+        },
+        {
+            'symbol': 'BBBBNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'BBB', 'weight': 500000, 'balance': '1e21'},
+                {'symbol': 'BNT', 'weight': 500000, 'balance': '1e21'},
+            ]
+        },
+        {
+            'symbol': 'CCCBNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'CCC', 'weight': 500000, 'balance': '1e21'},
+                {'symbol': 'BNT', 'weight': 500000, 'balance': '1e21'},
+            ]
+        },
+        {
+            'symbol': 'AAABNTBNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'AAABNT', 'weight': 500000, 'balance': '1e18'},
+                {'symbol': 'BNT', 'weight': 500000, 'balance': '1e21'},
+            ]
+        },
+        {
+            'symbol': 'BBBBNTBNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'BBBBNT', 'weight': 500000, 'balance': '1e18'},
+                {'symbol': 'BNT', 'weight': 500000, 'balance': '1e21'},
+            ]
+        },
+        {
+            'symbol': 'DDDAAABNTBNT', 'decimals': 18, 'fee': 1000, 'reserves': [
+                {'symbol': 'DDD', 'weight': 500000, 'balance': '1e21'},
+                {'symbol': 'AAABNTBNT', 'weight': 500000, 'balance': '1e15'},
+            ]
+        },
+    ]
+};
+
+async function getSymbol(tokenAddress) {
+    if (tokenAddress == ETH_RESERVE_ADDRESS)
```

Comment:
you sure this comparison is fine in regards to case sensitivity?

Reply:
@yudilevi:
- In truffle tests (web3 v0.x), all addresses are lower-cased (by either truffle or web3)
- In standalone scripts (web3 v1.x), we are using `Web3.utils.toChecksumAddress`

Reply:
I think that it's probably best practice to not rely on truffle for that

Reply:
Better than nothing...